laws/src/main/scala/cats/laws/TraverseFilterLaws.scala

```diff
@@ -52,6 +52,14 @@ trait TraverseFilterLaws[F[_]] extends FunctorFilterLaws[F] {
     G: Monad[G]
   ): IsEq[G[F[B]]] =
     fa.traverseEither(a => f(a).map(_.toRight(e)))((_, _) => Applicative[G].unit) <-> fa.traverseFilter(f)
+
+  def traverseCollectRef[G[_], A, B](
```

Comment:
this is the law we want, but adding it here isn't quite enough. It also has to be added to 

https://github.com/typelevel/cats/blob/a226d0121dd3b554d1639411a392c20e38c6addb/laws/src/main/scala/cats/laws/discipline/TraverseFilterTests.scala#L60

unfortunately (it is unfortunate that two separate files need to be editted, IMO).

Reply:
Added, thanks for review!